Sources/Subs-Auth.php

```diff
@@ -49,15 +49,26 @@ function setLoginCookie($cookie_length, $id, $password = '')
 		// Out with the old, in with the new!
 		if (isset($array[3]) && $array[3] != $cookie_state)
 		{
-			$cookie_url = url_parts($array[3] & 1 > 0, $array[3] & 2 > 0);
-			smf_setcookie($cookiename, $smcFunc['json_encode'](array(0, '', 0)), time() - 3600, $cookie_url[1], $cookie_url[0]);
+			// Maybe we need to temporarily pretend to be using local cookies
+			if ($cookie_state == 0 && $array[3] == 1)
+				$cookie_url = url_parts(true, false);
+			else
+				$cookie_url = url_parts($array[3] & 1 > 0, $array[3] & 2 > 0);
+
+			smf_setcookie($cookiename, $smcFunc['json_encode'](array(0, '', 0, $cookie_state), JSON_FORCE_OBJECT), 1, $cookie_url[1], $cookie_url[0]);
 		}
 	}
 
 	// Get the data and path to set it on.
-	$data = $smcFunc['json_encode'](empty($id) ? array(0, '', 0) : array($id, $password, time() + $cookie_length, $cookie_state));
+	$data = empty($id) ? array(0, '', 0, $cookie_state) : array($id, $password, time() + $cookie_length, $cookie_state);
 	$cookie_url = url_parts(!empty($modSettings['localCookies']), !empty($modSettings['globalCookies']));
 
+	// Allow mods to add custom info to the cookie
+	$custom_data = array();
+	call_integration_hook('integrate_cookie_data', array($data, &$custom_data));
```

Comment:
Integration hook doesn't work,
when you got the regexp on the otherside which not allow addional stuff.